pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -146,118 +146,245 @@ func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStart
 
 }
 
-// shouldEnsureSuggested checks if the exempt priority level exists and returns
-// whether the suggested flow schemas and priority levels should be ensured.
-func shouldEnsureSuggested(flowcontrolClientSet flowcontrolclient.FlowcontrolV1beta1Interface) (bool, error) {
-	if _, err := flowcontrolClientSet.PriorityLevelConfigurations().Get(context.TODO(), flowcontrol.PriorityLevelConfigurationNameExempt, metav1.GetOptions{}); err != nil {
-		if apierrors.IsNotFound(err) {
-			return true, nil
-		}
-		return false, err
+const thisFieldManager = "api-priority-and-fairness-config-producer-v1"
+
+// fsAutoUpdateStrategyFunc accepts the current and desired FlowSchema configuration and determines whether an update is necessary.
+// current is the existing configuration stored.
+// desired is the desired spec we want stored.
+// On return, shouldUpdate is set to true if auto update is required and newObject will represent the new configuration to be stored.
+type fsAutoUpdateStrategyFunc func(current, desired *flowcontrolv1beta1.FlowSchema) (newObject *flowcontrolv1beta1.FlowSchema, shouldUpdate bool)
+
+// plAutoUpdateStrategyFunc accepts the current and desired PriorityLevelConfiguration configuration and determines whether an update is necessary.
+// current is the existing configuration stored.
+// desired is the desired spec we want stored.
+// On return, shouldUpdate is set to true if auto update is required and newObject will represent the new configuration to be stored.
+type plAutoUpdateStrategyFunc func(current, desired *flowcontrolv1beta1.PriorityLevelConfiguration) (*flowcontrolv1beta1.PriorityLevelConfiguration, bool)
+
+// auto-update strategy for the suggested FlowSchema objects
+func fsSuggestedUpdateStrategy(current, desired *flowcontrolv1beta1.FlowSchema) (*flowcontrolv1beta1.FlowSchema, bool) {
+	if current == nil || desired == nil {
```

Comment:
Looking at the call site, the condition tested here will never be true.

Reply:
the test feeds `nil` for `current` to simulate a condition where the configuration does not exist in the cluster.